src/NHibernate/Impl/AbstractQueryImpl.cs

```diff
@@ -265,7 +266,10 @@ private string ExpandParameterList(string query, string name, TypedValue typedLi
 
 			var paramPrefix = isJpaPositionalParam ? StringHelper.SqlParameter : ParserHelper.HqlVariablePrefix;
 
-			return StringHelper.Replace(query, paramPrefix + name, string.Join(StringHelper.CommaSpace, aliases), true);
+			return Regex.Replace(
+				query,
+				$@"{paramPrefix}{name}\b",
```

Comment:
Shouldn't this place and others be wrapped in  `Regex.Escape` like `Regex.Escape(paramPrefix+name)`?

Reply:
This is unlikely that parameter name will contain any special regex characters.